### ISSUE TITLE
RA reporting "expected 1 argument, found 0" for `println!`

### ISSUE THREAD

**Issue by mi2ebi** [rust-lang/rust-analyzer#20675]
**rust-analyzer version**: rust-analyzer 1.91.0-nightly (fe55364 2025-08-29)

**rustc version**: rustc 1.91.0-nightly (fe5536432 2025-08-29)

**editor or extension**: Helix, https://github.com/helix-editor/helix

**relevant settings**: I've overridden the `check` command as my alias `cargo xnitpick`, which is
```toml
[alias]
xnitpick = """clippy --message-format=json-diagnostic-rendered-ansi --
  
  -W clippy::pedantic
  -A clippy::must_use_candidate

  -W clippy::nursery
  -A clippy::cognitive_complexity

  -W clippy::decimal_literal_representation
  -W clippy::deref_by_slicing
  -W clippy::if_then_some_else_none
  -W clippy::lossy_float_literal
  -W clippy::missing_asserts_for_indexing
  -W clippy::needless_raw_strings
  -W clippy::precedence_bits
  -W clippy::rest_pat_in_fully_bound_structs
  -W clippy::same_name_method
  -W clippy::semicolon_inside_block
  -W clippy::unused_trait_names
  
"""
```

**repository link (if public, optional)**: n/a

**code snippet to reproduce**:
```rust
fn main() {
    println!("the");
    println!("{x} is a cool number", x = 5);
}
```

<img width="520" height="165" alt="Image" src="https://github.com/user-attachments/assets/5543cb7d-1c4c-480d-aee9-3595a4c80dfd" />

`cargo build` etc do not emit any diagnostic for this

**Comment by ChayimFriedman2** [comment]
I can't reproduce this. Does this also reproduce on stable?

**Comment by mi2ebi** [comment]
Hm, I did `rustup default stable` and restarted RA and the diagnostics went away, then when I did `rustup default nightly` again and restarted they didn't come back

**Comment by mi2ebi** [comment]
oh, I changed the file and now it does

**Comment by ShoyuVanilla** [comment]
Might have `t-libs` changed `println!` or `format_args!` implementation again?

**Comment by ChayimFriedman2** [comment]
That could be the problem but then why can't I reproduce it on nightly?

**Comment by ChayimFriedman2** [comment]
@mi2ebi Are you still able to reproduce this? If yes, does it reproduce even after you upgrade your nightly?

**Comment by mi2ebi** [comment]
my nightly updated when i switched to 1.90 for stable and it hasn't happened since

**Comment by ChayimFriedman2** [comment]
Closing then.
